From dd3a72382123184c2786f784c6168b2d38e55a31 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 1 Feb 2016 13:08:24 +0700
Subject: [PATCH 08/14] 
 system-core_init-Re-introduce-lpm_boot-for-offline-charging02.patch

Change-Id: I0e7228a6ef1dab804a94c3fe78bfc2f890590b0b
---
 init/init.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/init/init.c b/init/init.c
index 7a68e62..ea24e53 100644
--- a/init/init.c
+++ b/init/init.c
@@ -72,6 +72,11 @@ static int   bootchart_count;
 #define BOARD_CHARGING_CMDLINE_VALUE "true"
 #endif
 
+#ifndef BOARD_LPM_BOOT_ARGUMENT_NAME
+#define BOARD_LPM_BOOT_ARGUMENT_NAME "lpm_boot"
+#define BOARD_LPM_BOOT_ARGUMENT_VALUE "1"
+#endif
+
 static char console[32];
 static char bootmode[32];
 static char hardware[32];
@@ -99,6 +104,8 @@ static time_t process_needs_restart;
 
 static const char *ENV[32];
 
+static unsigned lpm_bootmode = 0;
+
 static unsigned charging_mode = 0;
 
 /* add_environment - add "key=value" to the current environment */
@@ -736,6 +743,12 @@ static void import_kernel_nv(char *name, int for_emulator)
 
     if (!strcmp(name,"qemu")) {
         strlcpy(qemu, value, sizeof(qemu));
+#ifdef BOARD_LPM_BOOT_ARGUMENT_NAME
+    } else if (!strcmp(name,BOARD_LPM_BOOT_ARGUMENT_NAME)) {
+        if (!strcmp(value,BOARD_LPM_BOOT_ARGUMENT_VALUE)) {
+            lpm_bootmode = 1;
+        }
+#endif
     } else if (!strcmp(name,BOARD_CHARGING_CMDLINE_NAME)) {
         strlcpy(battchg_pause, value, sizeof(battchg_pause));
     } else if (!strncmp(name, "androidboot.", 12) && name_len > 12) {
@@ -1011,7 +1024,7 @@ static void selinux_initialize(void)
 static int charging_mode_booting(void)
 {
 #ifndef BOARD_CHARGING_MODE_BOOTING_LPM
-    return 0;
+    return lpm_bootmode;
 #else
     int f;
     char cmb;
-- 
2.5.0

